src/overlays/actors/ovl_En_Test6/z_en_test6.c

```diff
@@ -35,50 +53,1278 @@ const ActorInit En_Test6_InitVars = {
+            D_80A94910[i].unk_14 = func_80A90FC0;
+        }
+        phi_f24 += 50.0f;
+    }
+}
+
+void func_80A90C08(s16 arg0) {
+    func_8016566C(arg0);
+}
+
+void func_80A90C34(void) {
+    func_80165690();
+}
+
+void func_80A90C54(GlobalContext* globalCtx, f32 arg1) {
+    globalCtx->envCtx.unk_E5 = 1;
+    globalCtx->envCtx.unk_E6[0] = 250;
+    globalCtx->envCtx.unk_E6[1] = 250;
+    globalCtx->envCtx.unk_E6[2] = 250;
+    globalCtx->envCtx.unk_E6[3] = 255.0f * arg1;
+}
+
+void func_80A90D20(GlobalContext* globalCtx) {
+    globalCtx->envCtx.unk_E5 = 0;
+}
+
+void func_80A90D34(EnTest6* this, GlobalContext* globalCtx, EnTest6Struct* ptr) {
+    s32 pad;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    if (ptr->unk_00 != 0) {
+        Matrix_InsertTranslation(ptr->unk_08 * ptr->unk_04, ptr->unk_0C, ptr->unk_10 * ptr->unk_04, MTXMODE_NEW);
+        Matrix_Scale(ptr->unk_04 * 0.02f, ptr->unk_04 * 0.02f, ptr->unk_04 * 0.02f, MTXMODE_APPLY);
+        POLY_OPA_DISP = func_801660B8(globalCtx, POLY_OPA_DISP);
+        POLY_OPA_DISP = func_8012C724(POLY_OPA_DISP);
+
+        gSPSegment(POLY_OPA_DISP++, 0x08, Lib_SegmentedToVirtual(D_80A9402C[ptr->unk_00]));
+        gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPDisplayList(POLY_OPA_DISP++, gameplay_keep_DL_05F6F0);
+    }
+
+    Matrix_InsertTranslation(ptr->unk_08 * ptr->unk_04, ptr->unk_0C, ptr->unk_10 * ptr->unk_04, MTXMODE_NEW);
+    Matrix_Scale(2.0f * ptr->unk_04, 2.0f * ptr->unk_04, 2.0f * ptr->unk_04, MTXMODE_APPLY);
+    Matrix_InsertMatrix(&globalCtx->billboardMtxF, MTXMODE_APPLY);
+    Matrix_InsertZRotation_s(globalCtx->state.frames * 512, MTXMODE_APPLY);
+    Matrix_InsertTranslation(0.0f, 0.0f, 2.0f, MTXMODE_APPLY);
+
+    gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+
+    func_8012C2DC(globalCtx->state.gfxCtx);
+
+    gDPSetPrimColor(POLY_XLU_DISP++, 0x80, 0x80, 210, 210, 230, 128);
+    gDPSetEnvColor(POLY_XLU_DISP++, 255, 255, 255, 0);
+    gSPClearGeometryMode(POLY_XLU_DISP++, G_FOG | G_LIGHTING);
+    gSPDisplayList(POLY_XLU_DISP++, gameplay_keep_DL_054A90);
+    gSPSetGeometryMode(POLY_XLU_DISP++, G_FOG | G_LIGHTING);
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+}
+
+void func_80A90FC0(EnTest6* this, GlobalContext* globalCtx, EnTest6Struct* ptr) {
+    s32 pad;
+    Gfx* gfx = GRAPH_ALLOC(globalCtx->state.gfxCtx, sizeof(Gfx) * 2);
+    Gfx* gfx2 = gfx;
+    Hilite* sp70;
+    Vec3f sp64;
+
+    sp64.x = ptr->unk_08 * ptr->unk_04;
+    sp64.y = ptr->unk_0C;
+    sp64.z = ptr->unk_10 * ptr->unk_04;
+
+    sp70 = func_800BCBF4(&sp64, globalCtx);
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    if (gfx != NULL) {
+        func_8012C28C(globalCtx->state.gfxCtx);
+
+        gDPSetTileSize(gfx++, 1, sp70->h.x1 & 0xFFFF, sp70->h.y1 & 0xFFFF, (sp70->h.x1 + 60) & 0xFFFF,
+                       (sp70->h.y1 + 60) & 0xFFFF);
```

Comment:
Would `(u16)` work here instead of `& 0xFFFF`

Reply:
It doesn't, these all need a lw and not a lhu.